samwise.js

```diff
@@ -2,17 +2,15 @@
  * Autoplay plugin - Automatically advance slideshow after N seconds
```

Comment:
SCRIPT-This function seems to have grown quite complex. Can we break it down further?